tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -1151,6 +1151,66 @@ fillSymbolInfo(CursorSymbolInfo &Symbol, const DeclInfo &DInfo,
   return llvm::Error::success();
 }
 
+// If E is a literal, returns the declaration that should be reported by
+// cursor info for that initializer.
+static ValueDecl *getCursorInfoDeclForLiteral(Expr *E) {
+  if (auto *CollectionLit = dyn_cast<CollectionExpr>(E)) {
+    return CollectionLit->getInitializer().getDecl();
+  }
+
+  LiteralExpr* LitExpr = dyn_cast<LiteralExpr>(E);
+  if (!LitExpr) {
+    return nullptr;
+  }
+
+  bool IsObjectLiteral = dyn_cast<ObjectLiteralExpr>(E);
```

Comment:
Instead of `dyn_cast` and then relying on pointer to boolean conversions, I would use `isa` here.
```suggestion
  bool IsObjectLiteral = isa<ObjectLiteralExpr>(E);
```